Source/ChilliSource/Core/State/State.cpp

```diff
@@ -107,30 +121,32 @@ namespace ChilliSource
             system->OnFixedUpdate(in_fixedTimeSinceLastUpdate);
         }
         
-        m_scene->FixedUpdateEntities(in_fixedTimeSinceLastUpdate);
+        for(auto scene : m_scenes)
+        {
+            scene->FixedUpdateEntities(in_fixedTimeSinceLastUpdate);
+        }
         
         OnFixedUpdate(in_fixedTimeSinceLastUpdate);
     }
     //-----------------------------------------
     //-----------------------------------------
-    void State::RenderSnapshot(class RenderSnapshot& in_renderSnapshot) noexcept
+    void State::RenderSnapshot(class RenderSnapshot& renderSnapshot, IAllocator* frameAllocator) noexcept
     {
         for(auto& system : m_systems)
         {
-            system->OnRenderSnapshot(in_renderSnapshot);
+            system->OnRenderSnapshot(renderSnapshot, frameAllocator);
         }
-        
-        m_scene->RenderSnapshotEntities(in_renderSnapshot);
-        
-        OnRenderSnapshot(in_renderSnapshot);
```

Comment:
I removed it because of the ambiguity surrounding multiple render targets (the confusion still kind of exists for systems). It's not obvious whether this is called for all render snapshots or just the main one. Should we call it for all and have some way of distinguising the render targets?